homeassistant/components/filter/config_flow.py

```diff
@@ -0,0 +1,211 @@
+            ),
+        }
+
+    elif handler.options[CONF_FILTER_NAME] == FILTER_NAME_TIME_SMA:
+        filter_schema = {
+            vol.Optional(CONF_TIME_SMA_TYPE, default=TIME_SMA_LAST): SelectSelector(
+                SelectSelectorConfig(
+                    options=[TIME_SMA_LAST],
+                    mode=SelectSelectorMode.DROPDOWN,
+                    translation_key=CONF_TIME_SMA_TYPE,
+                )
+            ),
+            vol.Required(CONF_FILTER_WINDOW_SIZE): DurationSelector(
+                DurationSelectorConfig(enable_day=False, allow_negative=False)
+            ),
+        }
+
+    elif handler.options[CONF_FILTER_NAME] == FILTER_NAME_THROTTLE:
+        filter_schema = {
+            vol.Optional(
+                CONF_FILTER_WINDOW_SIZE, default=DEFAULT_WINDOW_SIZE
+            ): NumberSelector(
+                NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+            ),
+        }
+
+    elif handler.options[CONF_FILTER_NAME] == FILTER_NAME_TIME_THROTTLE:
+        filter_schema = {
+            vol.Required(CONF_FILTER_WINDOW_SIZE): DurationSelector(
+                DurationSelectorConfig(enable_day=False, allow_negative=False)
+            ),
+        }
+
+    base_schema: dict[vol.Marker, Selector] = {
+        vol.Optional(CONF_FILTER_PRECISION, default=DEFAULT_PRECISION): NumberSelector(
+            NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
+        )
+    }
+
+    return vol.Schema({**filter_schema, **base_schema})
+
+
+async def validate_options(
+    handler: SchemaCommonFlowHandler, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate options selected."""
+
+    if CONF_FILTER_WINDOW_SIZE in user_input and isinstance(
+        user_input[CONF_FILTER_WINDOW_SIZE], float
+    ):
+        user_input[CONF_FILTER_WINDOW_SIZE] = int(user_input[CONF_FILTER_WINDOW_SIZE])
+    if CONF_FILTER_TIME_CONSTANT in user_input:
+        user_input[CONF_FILTER_TIME_CONSTANT] = int(
+            user_input[CONF_FILTER_TIME_CONSTANT]
+        )
+    if CONF_FILTER_PRECISION in user_input:
+        user_input[CONF_FILTER_PRECISION] = int(user_input[CONF_FILTER_PRECISION])
+
+    handler.parent_handler._async_abort_entries_match({**handler.options, **user_input})  # noqa: SLF001
+
+    return user_input
+
+
+DATA_SCHEMA_SETUP = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): TextSelector(),
+        vol.Required(CONF_ENTITY_ID): EntitySelector(
+            EntitySelectorConfig(domain=[SENSOR_DOMAIN])
+        ),
+        vol.Required(CONF_FILTER_NAME): SelectSelector(
+            SelectSelectorConfig(
+                options=FILTERS,
+                mode=SelectSelectorMode.DROPDOWN,
+                translation_key=CONF_FILTER_NAME,
+            )
+        ),
+    }
+)
+
+
+CONFIG_FLOW = {
+    "user": SchemaFlowFormStep(
+        schema=DATA_SCHEMA_SETUP,
+        next_step="options",
+    ),
```

Comment:
Cant we have different steps for different types of filters? Not sure it maps to the old config structure, but feels it might be clearer?

Reply:
Not sure what the benefit would be?
Feels like it would just end up with duplicating a lot of code.

Reply:
Because you have a if type equal checks in the option flow schema constru6

Reply:
Right, I think I see your point. I'll make an adjustment here! 👍